### PR TITLE
Single user server

### PR DESCRIPTION
**Pull request by MitchBears**
Simple REST API for adding, changing, and deleting patterns for a given user. Run tests with `npm run mocha`. Next step would be to implement the save button for a single user so that changes to a single pattern persist.

### CODE REVIEW

File: tests/server.test.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  } else if (!_.isEqual(pattern1.height, pattern2.height)) {
+    return false;
+  } else if (!_.isEqual(pattern1.width, pattern2.width)) {
+    return false;
+  }
+  return true;
+};
+
+// PUT method
+describe('PUT method', () => {
```
Comment (zachBears) [inline tests/server.test.js]: For what route? Make your describes more clear.

File: tests/server.test.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  it('should return the pattern whose creator id is the same as User 1s googleId', (done) => {
```
Comment (zachBears) [inline tests/server.test.js]: You should wrap this in another describe that says "when I request the first pattern" or something like that. Same below.

File: tests/server.test.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        if (err) {
+          throw err;
+        }
+        if (testPatternEquality(res.body, patterns[0])) {
+          done();
+        }
+        throw new Error('Incorrect pattern saved');
+      });
+  });
+  it('should return the second pattern whose creator id is the same as User 1s googleId', (done) => {
```
Comment (zachBears) [inline tests/server.test.js]: What about error cases? Invalid userID/pattern?

File: tests/server.test.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+          done();
+        }
+        throw new Error('Incorrect pattern saved');
+      });
+  });
+});
+
+describe('GET methods', () => {
+  describe('/user', () => {
+    it('should confirm that user has two patterns', (done) => {
```
Comment (zachBears) [inline tests/server.test.js]: These its should all be wrapped in another describe to add more context. Something like "when I access a user with two patterns"

File: tests/server.test.js [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+          if (err) {
+            throw err;
+          }
+          if (res.body.length === 2) {
+            done();
+          }
+          throw new Error('Two patterns are not saved');
+        });
+    });
+    it('should return a 404 error due to incorrect googleId', (done) => {
```
Comment (zachBears) [inline tests/server.test.js]: Good.

File: tests/server.test.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+          if (err) {
+            throw err;
+          }
+          if (testPatternEquality(res.body, patterns[0])) {
+            done();
+          }
+          throw new Error('Incorrect pattern returned.');
+        });
+    });
+    it('should return a 404 error due to index out of range', (done) => {
```
Comment (zachBears) [inline tests/server.test.js]: Good.

File: tests/server.test.js [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+            throw err;
+          }
+          done();
+        });
+    });
+  });
+});
+
+describe('DELETE method', () => {
+  it('should return the users patterns without the first pattern', (done) => { // eslint-disable-line consistent-line
```
Comment (zachBears) [inline tests/server.test.js]: Why is this disabled?

Comment (zachBears) [inline tests/server.test.js]: once again. wrap in describe that says "when deleting a pattern"

File: server.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        throw findError;
+      }
+      if (!data) {
+        return res.status(404).send();
+      }
+      return res.send(data[0]);
+    });
+  });
+});
+app.get('/user', (req, res) => {
```
Comment (zachBears) [inline server.js]: space above

File: server.js [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+          }
+        });
+        return res.send(doc[0].patterns);
+      }
+      return res.status(500).send(new Error('Cannot delete pattern that doesnt exist.'));
+    });
+});
+
+app.listen(8081, () => {
+});
```
Comment (zachBears) [inline server.js]: Use the identity instead of this function _ => _

File: package.json [hunk 10]
```diff
@@ -3,36 +3,50 @@
   "version": "0.0.1",
   "author": "Mitchell Bears",
   "scripts": {
-    "start": "webpack-dev-server --progress",
+    "start": "nodemon server.js",
     "build": "NODE_ENV=production webpack -p --progress --colors",
     "lint": "./scripts/lint.sh",
-    "test": "./scripts/lint.sh"
+    "test": "./scripts/lint.sh",
```
Comment (MitchBears) [inline package.json]: The automated tester does not have access to a mongo database so I believe the tests will always fail.

**Comment by MitchBears** [comment]
@zbears Review changes addressed.

**Pull request merged**